src/base/bittorrent/torrentimpl.cpp

```diff
@@ -112,8 +112,8 @@ namespace
         QString firstTrackerMessage;
         QString firstErrorMessage;
 #if (LIBTORRENT_VERSION_NUM >= 20000)
-        const size_t numEndpoints = nativeEntry.endpoints.size() * ((hashes.has_v1() && hashes.has_v2()) ? 2 : 1);
-        trackerEntry.endpoints.reserve(static_cast<decltype(trackerEntry.endpoints)::size_type>(numEndpoints));
```

Comment:
>Use proper signed int type

Sorry, can't understand it...
`trackerEntry.endpoints` is `QVector`. `QVector::size_type` is `typedef` for `int` in Qt5.
I don't use `int` explicitly since it is changed in Qt6 to be `qsizetype` so the code should stay valid there.

Reply:
>I don't use int explicitly since it is changed in Qt6 to be qsizetype so the code should stay valid there.

https://doc.qt.io/qt-5/qtglobal.html#qsizetype-alias
`qsizetype` is signed number type, so it is incorrect to use `size_t numEndpoints`.
I changed to `qsizetype` directly, should be better now.

Reply:
Now you pass `qsizetype` in `QVector::reserve()` that accepts `int` in Qt5.
Seems you still require decltype way to produce compatible code:
```c++
const auto numEndpoints = static_cast<decltype(trackerEntry.endpoints)::size_type>(nativeEntry.endpoints.size() * ((hashes.has_v1() && hashes.has_v2()) ? 2 : 1));
trackerEntry.endpoints.reserve(numEndpoints);
```

Reply:
>Now you pass qsizetype in QVector::reserve() that accepts int in Qt5.

I pushed a fix.

>Seems you still require decltype way to produce compatible code:

I didn't take the suggestion because I think `numEndpoints` should stayed with signed number in all cases.

Reply:
Although if it doesn't produce any warnings, let's not complicate it.

Reply:
>I think numEndpoints should stayed with signed number in all cases.

How does my suggestion contradict this? `QVector::size_type` is signed in all cases.

Reply:
>How does my suggestion contradict this? QVector::size_type is signed in all cases.

Seems I misread your code and provided the wrong reason. Another reason/difference is using `decltype(trackerEntry.endpoints)::size_type` read as following the *implementation detail of a container* while using `qsizetype` directly states that we want it to be the type. Clearly we want the latter because the variable has additional meaning tied to it (number of endpoints).

Reply:
Ok.
But we can/should remove the `static_cast` in `reserve()` once we reach Qt6.